src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,166 @@
+                }
+                current = current.next;
+            }
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+            } else {
+                current.next = node;
+                size++;
+            }
+        }
+        if (size > loader) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (key == null) {
+            return getNullKey();
+        }
+        int index = getHash(key);
+        V value = null;
+        Node<K, V> currentNode = initialArray[index];
+        while (currentNode != null) {
+            if (currentNode.key != null && currentNode.key.equals(key)) {
+                value = currentNode.value;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        return value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        int newCapacity = initialArray.length * NUMBER_TO_MULTIPLY;
+        Node<K, V>[] newArray = new Node[newCapacity];
+        for (int i = 0; i < initialArray.length; i++) {
+            Node<K, V> current = initialArray[i];
+            while (current != null) {
+                Node<K, V> next = current.next;
+                int newIndex = current.hash % newCapacity;
+                current.next = newArray[newIndex];
+                newArray[newIndex] = current;
+                current = next;
+            }
+        }
+        initialArray = newArray;
+        threshold *= NUMBER_TO_MULTIPLY;
+    }
+
+    private V getNullKey() {
+        Node<K, V> currentNode = initialArray[ZERO];
+        while (currentNode != null) {
+            if (currentNode.key == null) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private void putNullKey(V value) {
+        if (initialArray[ZERO] == null) {
+            initialArray[ZERO] = newNode(null, value);
+            size++;
+        } else {
+            Node<K, V> current = initialArray[0];
+            while (current.next != null) {
+                if (current.key == null) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            if (current.key == null) {
+                current.value = value;
+            } else {
+                current.next = newNode(null, value);
+                size++;
+            }
+        }
+        if (size > threshold) {
+            resize();
+        }
+    }
+
+    private Node<K, V> newNode(K key, V value) {
+        return new Node<>(getHash(key), key, value, null);
+    }
+
+    private int getHash(K key) {
+        return key == null ? ZERO
+                : Math.abs(key.hashCode()) % DEFAULT_INITIAL_CAPACITY;
+    }
+
+    static class Node<K, V> {
```

Comment:
There should be some access modifier (private, public, etc)

Reply:
Agree